File: javascript/series/series.js
export class Series {
  constructor(series) {
    if (series.length === 0) {
      throw new Error('series cannot be empty');
    }
    this._series = series.split('').map(value => Number(value));
  }

  slices(sliceLength) {
    if (sliceLength === 0) {
      throw new Error('slice length cannot be zero');
    } else if (sliceLength < 0) {
      throw new Error('slice length cannot be negative');
    } else if (sliceLength > this._series.length) {
      throw new Error('slice length cannot be greater than series length');
    };
    let slices = [];
    for (let i = 0; i <= this._series.length - sliceLength; i++) {
      slices.push(this._series.slice(i, i + sliceLength));
    };
    return slices;
  }
}

File: python/series/series.py
def slices(series, length):
    if not series:
        raise ValueError("series cannot be empty")
    if length == 0:
        raise ValueError("slice length cannot be zero")
    if length < 0:
        raise ValueError("slice length cannot be negative")
    if length > len(series):
        raise ValueError("slice length cannot be greater than series length")
    return [series[index:index + length] for index in range(len(series) - length + 1)]
